File: pinyougou-web-shop/src/main/java/cn/itcast/core/controller/SeckillOrderController.java
package cn.itcast.core.controller;

import cn.itcast.core.pojo.good.Goods;
import cn.itcast.core.pojo.seckill.SeckillOrder;
import cn.itcast.core.service.SeckillOrderService;
import com.alibaba.dubbo.config.annotation.Reference;
import entity.PageResult;
import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@RequestMapping("/seckillOrder")
public class SeckillOrderController {
    @Reference
    private SeckillOrderService seckillOrderService;
    @RequestMapping("/findAll")
    public List<SeckillOrder> findAll(){
        String name = SecurityContextHolder.getContext().getAuthentication().getName();
     return    seckillOrderService.findAll(name);
    }

    @RequestMapping("/search")
    public PageResult search(Integer page, Integer rows, @RequestBody SeckillOrder seckillOrder){
        String name = SecurityContextHolder.getContext().getAuthentication().getName();
        seckillOrder.setUserId(name);
        return seckillOrderService.search(page,rows,seckillOrder);
    }

}

File: pinyougou-web-shop/src/main/webapp/js/controller/seckillOrderController.js
// 定义控制器:
app.controller("seckillOrderController",function($scope,$controller,$http,seckillOrderService){
	// AngularJS中的继承:伪继承
	$controller('baseController',{$scope:$scope});
	
	// 查询所有的品牌列表的方法:
	$scope.findAll = function(){
		// 向后台发送请求:
        seckillOrderService.findAll().success(function(response){
			$scope.list = response;
		});
	}
    //搜索
    $scope.searchEntity = {};
    $scope.search=function(page,rows){
        seckillOrderService.search(page,rows,$scope.searchEntity).success(
            function(response){

                $scope.list=response.rows;
                $scope.paginationConf.totalItems=response.total;//更新总记录数
            }
        );
    }


	
});
